# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../openoffice-org/patches/shell.diff
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Index: source/unix/sysshell/systemshell.cxx
===================================================================
RCS file: /cvs/gsl/shell/source/unix/sysshell/systemshell.cxx,v
retrieving revision 1.8
diff -u -r1.8 systemshell.cxx
--- shell/source/unix/sysshell/systemshell.cxx	19 Jun 2007 16:12:44 -0000	1.8
+++ shell/source/unix/sysshell/systemshell.cxx	13 Apr 2008 15:27:22 -0000
@@ -62,6 +62,8 @@
 #include <com/sun/star/uno/XComponentContext.hpp>
 #endif
 
+#include "uno/current_context.hxx"
+
 #include "osl/module.hxx"
 
 using namespace ::com::sun::star;
@@ -116,6 +118,25 @@
     
     bool init_recent_manager_api()
     {
+        bool bInGnome = false;
+        try
+        {
+            uno::Reference< uno::XCurrentContext > xCurrentContext(uno::getCurrentContext());
+
+            if (xCurrentContext.is())
+            {
+                uno::Any aValue = xCurrentContext->getValueByName(
+                    rtl::OUString( RTL_CONSTASCII_USTRINGPARAM( "system.desktop-environment" ) ) );
+
+                rtl::OUString aDesktopEnvironment;
+                if ( (aValue >>= aDesktopEnvironment) && (aDesktopEnvironment.equalsIgnoreAsciiCaseAscii("gnome")) )
+                    bInGnome = true;
+            }
+        } catch ( ... ) {}
+
+        if (!bInGnome)
+            return false;
+
         oslModule hDefault;
         if( osl_getModuleHandle( NULL, &hDefault ) )
         {
